README.md

```diff
@@ -69,10 +69,14 @@ Note that the external-provisioner does not scale with more replicas. Only one e
 
 * `--cloning-protection-threads <num>`: Number of simultaneously running threads, handling cloning finalizer removal. Defaults to `1`.
 
-* `--http-endpoint`: The TCP network address where the HTTP server for diagnostics, including metrics and leader election health check, will listen (example: `:8080` which corresponds to port 8080 on local host). The default is empty string, which means the server is disabled.
+* `--http-endpoint`: The TCP network address where the HTTP(S) server for diagnostics, including metrics and leader election health check, will listen (example: `:8080` which corresponds to port 8080 on local host). The default is empty string, which means the server is disabled.
 
 * `--metrics-path`: The HTTP path where prometheus metrics will be exposed. Default is `/metrics`.
 
+* `--http-endpoint-key-file`: The path to TLS key file used for running HTTPS server. The default is empty string, which means that TLS will not be used when running HTTP server. Both `--http-endpoint-key-file` and `--http-endpoint-cert-file` flags must be set to enable TLS support.
```

Comment:
Do we need a CA file as well? Who signed this http certificate?

Reply:
No, no need for CA file. The user must provide signed cert file.

Reply:
no need for CA file means this  cert is a self-signed cert?

Reply:
> In a typical public-key infrastructure (PKI) scheme, the certificate issuer is a certificate authority (CA), usually a company that charges customers to issue certificates for them.

So the CA issues the certificate and we just need to provide path to it. It can be self-signed cert (local CA).

I don't see a reason why would we need CA file? Could you explain that?

Reply:
Bare with me if my understand is not correct, I am not very good at certs. The provisioner itself will serve as a HTTPs server, so if there is a client want to talk to this server, the server need to verify the client cert is signed by the correct CA. So basically supplying it a CA file will tell the HTTPs server which CA to trust. This is my understanding, I might be wrong but happy to learn it

Reply:
Bare with me if my understand is not correct, I am not very good at certs. The provisioner itself will serve as a HTTPs server, so if there is a client want to talk to this server, the server need to verify the client cert is signed by the correct CA. So basically supplying it a CA file will tell the HTTPs server which CA to trust. This is my understanding, I might be wrong but happy to learn it

Reply:
cc @msau42 to help me with my understanding

Reply:
As far I understand this, the client doesn't need to provide the certs. The client certs work as authentication but this isn't in the scope of this MR. This MR deals with server certs. What this means is that the client issues a request and wants to check if the server is indeed valid, secure and trusted entity (not some kind of impostor).

> In SSL handshake process, a client makes a request for a secured connection, the server shows its public certificate for the acceptance. The client has already trusted certificate lists stored in called truststore. If the presented certificate matches with the prescribed list of certificates, an encrypted connection could be established.

Reply:
oh I see. So here in your configuration you did not enable client-side verification on the HTTPs server. So thats why the CA is not needed. I guess it is okay